src/test/java/org/opentripplanner/routing/core/AccessibilityRoutingTest.java

```diff
@@ -225,6 +236,69 @@ public void canApplyNoWheelchairAccessStreetPenalty() {
         assertThatPolylinesAreEqual("sz_mE`v}aO?@M?cA?}A?uFCM??L?R", leg.legGeometry.getPoints());
     }
 
+    @Test
+    public void canConfigureSlopeAvoidanceAndScore() {
+        double length = 650.0;
+        Coordinate[] profile = new Coordinate[] {
+                new Coordinate(0, 0), // slope = 0.04
+                new Coordinate(length / 2, length / 50.0),
+                new Coordinate(length, 0) // slope = -0.04
+        };
+        PackedCoordinateSequence elev = new PackedCoordinateSequence.Double(profile);
+
+        SlopeCosts costs = ElevationUtils.getSlopeCosts(elev, false);
+
+        // we check that we really are going to set a 4 % slope
+        assertEquals(0.04, costs.maxSlope);
+
+        // here we get Old Wheat St (https://www.openstreetmap.org/way/9270835)
+        // and set an artificial slope so we can test the feature
+        Envelope e = new Envelope(new Coordinate(-84.36795, 33.75665));
+        e.expandBy(0.001);
+        graph.streetIndex.getEdgesForEnvelope(e).stream()
+                .filter(StreetWithElevationEdge.class::isInstance)
+                .map(StreetWithElevationEdge.class::cast)
+                .filter(s -> s.getName().startsWith("Old Wheat St"))
+                .forEach(s -> s.setElevationProfile(elev, false));
+
+        GenericLocation start = new GenericLocation(33.75561, -84.36798);
+        GenericLocation end = new GenericLocation(33.75573, -84.36701);
+
+        // avoids Old Wheat St as it's too steep
+        Itinerary i = getTripPlan(start, end, r -> {
+            r.setMode(TraverseMode.WALK);
+            r.maxSlope = 0.03;
+            r.wheelchairMaxSlopeExceededReluctance = 5;
+        }).itinerary.get(0);
+
+        Leg leg = i.legs.get(0);
+        assertEquals("WALK", leg.mode);
+        assertThatPolylinesAreEqual("q{_mE|b}aOS?SACU?AeCmDr@@dBD??", leg.legGeometry.getPoints());
+        assertEquals(1, leg.accessibilityScore);
+
+        // we go along Old Wheat St as we set a really high maxSlope
+        Itinerary i2 = getTripPlan(start, end, r -> {
+            r.setMode(TraverseMode.WALK);
+            r.maxSlope = 1;
+        }).itinerary.get(0);
+        Leg leg2 = i2.legs.get(0);
+        assertThatPolylinesAreEqual("q{_mE|b}aOS?SAJU@I@}C??", leg2.legGeometry.getPoints());
+        assertEquals(1, leg2.accessibilityScore);
+        assertEquals("WALK", leg2.mode);
+
+        // we go along Old Wheat St as we set a really low reluctance
+        Itinerary i3 = getTripPlan(start, end, r -> {
+            r.setMode(TraverseMode.WALK);
+            r.maxSlope = 0.03;
+            r.wheelchairMaxSlopeExceededReluctance = 1;
+        }).itinerary.get(0);
+        Leg leg3 = i3.legs.get(0);
+        assertThatPolylinesAreEqual("q{_mE|b}aOS?SAJU@I@}C??", leg3.legGeometry.getPoints());
+        assertEquals(0.9f, leg3.accessibilityScore);
```

Comment:
I don't understand how itinerary `i1` with the same `maxSlope` but a higher max slope reluctance gets a higher (better) accessibility score than itinerary `i3`. Can you explain?

Reply:
Sure. In i1 we have a high reluctance so we avoid the steep road: https://tinyurl.com/y2az95mo
This means we use flat streets leading to a good accessibility score.

In i3 we have a low reluctance and don't avoid the steep part: https://tinyurl.com/y3bge8u8
Because we traverse a steep edge (Old Wheat St) we exceed the maxSlope leading to no cost penalty but a lower accessibility score.

In summary: those two use different streets to reach the destination.

Does this make sense or have I understood how this is meant to work?

Reply:
Ok, thanks for the clarification!

Reply:
You're right though: In practise there isn't really a reason to have a low reluctance. I just lowered it in this test so that the we force a route that takes a steep road.